From: James Bottomley <James.Bottomley@HansenPartnership.com>
Subject: oid_registry: allow arbitrary size OIDs
Date: Wed, 26 Nov 2025 14:22:33 -0500

tidy up perl with better my placement also set bc to arbitrary size

Link: https://lkml.kernel.org/r/dbc90c344c691ed988640a28367ff895b5ef2604.camel@HansenPartnership.com
Signed-off-by: James Bottomley <James.Bottomley@HansenPartnership.com>
Cc: Blaise Boscaccy <bboscaccy@linux.microsoft.com>
Cc: David Howells <dhowells@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/lib/build_OID_registry~oid_registry-allow-arbitrary-size-oids-v2
+++ a/lib/build_OID_registry
@@ -60,7 +60,7 @@ for (my $i = 0; $i <= $#names; $i++) {
     # Determine the encoded length of this OID
     my $size = $#components;
     for (my $loop = 2; $loop <= $#components; $loop++) {
-	$ENV{'BC_LINE_LENGTH'} = "200";
+	$ENV{'BC_LINE_LENGTH'} = "0";
 	my $c = `echo "ibase=10; obase=2; $components[$loop]" | bc`;
 	chomp($c);
 
@@ -103,7 +103,7 @@ for (my $i = 0; $i <= $#names; $i++) {
 
     for (my $loop = 2; $loop <= $#components; $loop++) {
 	# get the base 2 representation of the component
-	$ENV{'BC_LINE_LENGTH'} = "200";
+	$ENV{'BC_LINE_LENGTH'} = "0";
 	my $c = `echo "ibase=10; obase=2; $components[$loop]" | bc`;
 	chomp($c);
 
@@ -114,15 +114,12 @@ for (my $i = 0; $i <= $#names; $i++) {
 	$c = substr("0000000", 0, ($tmp + 1) * 7 - length($c)).$c;
 
 	# Base128 encode the number
-	my $j;
-	my $b;
-	for ($j = 0; $j < $tmp; $j++) {
-	    $b = oct("0b".substr($c, $j * 7, 7));
+	for (my $j = 0; $j < $tmp; $j++) {
+	    my $b = oct("0b".substr($c, $j * 7, 7));
 
 	    push @octets, $b | 0x80;
 	}
-	$b = oct("0b".substr($c, $j * 7, 7));
-	push @octets, $b;
+	push @octets, oct("0b".substr($c, $tmp * 7, 7));
     }
 
     push @encoded_oids, \@octets;
_
